elfpy/utils/apeworx_integrations.py

```diff
@@ -0,0 +1,250 @@
+
+
+def get_transaction_trade_event(tx_receipt: Receipt) -> ContractLog:
+    r"""Parse the transaction receipt to get the trade event
+
+    Arguments
+    ---------
+    tx_receipt: ape_ethereum.transactions.Receipt
+        `Ape transaction receipt
+        <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.transactions.ReceiptAPI>`_
+
+    Returns
+    -------
+    ape.types.ContractLog
+        The primary emitted trade (a "TransferSingle" `ContractLog
+        <https://docs.apeworx.io/ape/stable/methoddocs/types.html#ape.types.ContractLog>`_
+        ) event, excluding periferal events.
+    """
+    single_events: list[ContractLog] = []
+    for tx_event in tx_receipt.events:
+        if tx_event.event_name == "TransferSingle":
+            single_events.append(tx_event)
+    if len(single_events) != 1:
+        raise ValueError(f'The transaction receipt should have one "TransferSingle" event, not {len(single_events)}.')
+    return single_events[0]
+
+
+def ape_open_position(
+    trade_prefix: AssetIdPrefix,
+    hyperdrive_contract: ContractInstance,
+    agent_address: AccountAPI,
+    trade_amount: int,
+) -> tuple[dict[str, Any], Receipt]:
+    r"""Open a long trade on the Hyperdrive Solidity contract using apeworx.
+
+    Arguments
+    ---------
+    trade_prefix: AssetIdPrefix
+        IntEnum specifying whether the trade is a long (0) or a short (1).
+    hyperdrive_contract: ape.contracts.base.ContractInstance
+        Ape project `ContractInstance
+        <https://docs.apeworx.io/ape/stable/methoddocs/contracts.html#ape.contracts.base.ContractInstance>`_
+        wrapped around the initialized MockHyperdriveTestnet smart contract.
+    agent_address: ape.api.accounts.AccountAPI
+        Ape address container, or `AccountAPI
+        <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.accounts.AccountAPI>`_
+        representing the agent which is executing the action.
+    base_amount: int
+        Unsigned int-256 representation of the base amount that the agent wishes to provide.
+
+    Returns
+    -------
+    Tuple[dict[str, Any], ape_ethereum.transactions.Receipt]
+        A tuple containing an update dictionary for the Hyperdrive pool state
+        as well as the Ape-ethereum transaction `receipt
+        <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.transactions.ReceiptAPI>`_.
+    """
+    with ape.accounts.use_sender(agent_address):  # sender for contract calls
+        if trade_prefix == AssetIdPrefix.LONG:  # open a long
+            min_output = 0
+            as_underlying = False  # mockHyperdriveTestNet does not support as_underlying=True
+            tx_receipt = hyperdrive_contract.openLong(  # type: ignore
+                trade_amount,  # base
+                min_output,
+                agent_address,
+                as_underlying,
+            )
+        elif trade_prefix == AssetIdPrefix.SHORT:  # open a short
+            max_deposit = trade_amount
```

Comment:
this feels... strange.  the `max_deposit` is the `trade_amount`?  this is probably a temporary thing for now?  add a comment.  i guess this works as a calc_in_given_out kinda thing?  makes me think we'll probably want ape_open_short and ape_open_long since the latter has a `min_out` and the former will have a `max_in`